pkg/networkserver/downlink.go

```diff
@@ -627,8 +642,8 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context) error {
 							req.Rx1DataRateIndex = rx1DRIdx
 						}
 						if rx2 {
-							req.Rx2Frequency = dev.MACState.CurrentParameters.Rx2Frequency
-							req.Rx2DataRateIndex = dev.MACState.CurrentParameters.Rx2DataRateIndex
+							req.Rx2Frequency = dev.PendingMACState.CurrentParameters.Rx2Frequency
```

Comment:
Then you can also just use `macState` here and below

Reply:
I'm explicitly using the `dev.PendingMACState` here on purpose

Reply:
not applicable anymore